recipes/opencv/4.x/conanfile.py

```diff
@@ -504,7 +504,8 @@ def _configure_cmake(self):
                 "armv8": "aarch64",
                 "armv8.3": "aarch64",
             }.get(str(self.settings.arch), str(self.settings.arch))
-            self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = cmake_system_processor
+            if self.settings.os != "Android":
+                self._cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = cmake_system_processor
```

Comment:
```suggestion
            self._cmake.definitions["CONAN_OPENCV_SYSTEM_PROCESSOR"] = cmake_system_processor
```

and add this in CMakeLists wrapper after `conan_basic_setup`:

```cmake
if(NOT CMAKE_SYSTEM_PROCESSOR)
    set(CMAKE_SYSTEM_PROCESSOR ${CONAN_OPENCV_SYSTEM_PROCESSOR})
endif()
```